jenkins-setup/action.yml

```
@@ -0,0 +1,17 @@
+name: 'jenkins-setup'
+description: 'Install the Jenkins core, Jenkinsfile-runner, plugin-installation-manager-tool and the minimum required set of plugins for pipeline execution'
+inputs:
+  jenkins-version:
+    description: 'The version of jenkins core to download. Default is 2.346.1.'
+    default: 2.346.1
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'adopt'
+        java-version: '11'  
+    - name: Download Jenkins binaries
+      run: ${GITHUB_ACTION_PATH}/setup.sh ${{inputs.jenkins-version}}
```

Note: Do not commit this suggestion as it is

Instead of passing the Jenkins version to your shell script as a cmdline argument, you can set it as an environment variable. This will make the interface consistent. The other variables like `JENKINS_CORE_URL` can then also be **inputs** to the action with default values specified in `action.yml`. This should simplify the setup script and users wouldn't have to bother setting environment variables.

```suggestion
      run: ${GITHUB_ACTION_PATH}/setup.sh
      env:
        JENKINS_VERSION: ${{ inputs.jenkins-version }}
        JENKINS_CORE_URL: ${{ inputs.jenkins-core-url }}
        # and so on...
```

---

Can the users' envs override the envs in the `action.yml`?

---

@AbhyudayaSharma  I tried my best to implement the interface as you described. I think it's not a good practice to set the default values as templates for `JENKINS_PM_URL`, `JENKINS_CORE_URL` and `JENKINS_JFR_URL` in the inputs. So I use the full URL as default value. If the users want to use different version for them, script will still use the templates. If the users specify their own download center in the inputs, the version inputs will become invalid and I think it's natural. In a word, GitHub Action doesn't permit pass templates as input because it will invalidate escape words.